apps/test/unit/aichat/ChatEventLoggerTest.ts

```diff
@@ -0,0 +1,55 @@
+import * as aichatApi from '@cdo/apps/aichat/aichatApi';
+import ChatEventLogger from '@cdo/apps/aichat/chatEventLogger';
+import {AichatContext, ChatMessage} from '@cdo/apps/aichat/types';
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+
+describe('ChatEventLogger', () => {
+  let userChatMessage: ChatMessage;
+  let aichatContext: AichatContext;
+  let chatEventLogger: ChatEventLogger;
+  let postLogChatEventSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    userChatMessage = {
+      role: Role.USER,
+      chatMessageText: 'hello',
+      status: 'OK',
+      timestamp: Date.now(),
+    };
+    aichatContext = {
+      currentLevelId: 123,
+      scriptId: 321,
+      channelId: 'abc123',
+    };
+    chatEventLogger = ChatEventLogger.getInstance();
+    postLogChatEventSpy = jest
+      .spyOn(aichatApi, 'postLogChatEvent')
+      .mockResolvedValue({chat_event_id: 1, chat_event: userChatMessage});
+  });
+
+  afterEach(() => {
+    postLogChatEventSpy.mockRestore();
+  });
+
+  it('logChatEvent calls on postLogChatEvent', async () => {
+    chatEventLogger.logChatEvent(userChatMessage as ChatMessage, aichatContext);
+    expect(postLogChatEventSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('postLogChatEvent not called when sendingInProgress is true', async () => {
+    // Simulate a sending in progress state.
+    chatEventLogger.setSendingInProgress(true);
```

Comment:
Instead of manually setting `sendingInProgress` to true, could we stub the `postLogChatEvent` function to just wait for some time inside a promise? That would more realistically simulate a network request and allow us to actually test the logic of changing the flag while sending is in progress.

Reply:
Yes - good call! I knew there was probably a better way to test this. Thanks for your help!